strax/mailbox.py

```diff
@@ -467,13 +493,14 @@ def divide_outputs(source,
     # TODO: this code duplicates exception handling and cleanup
     # from Mailbox.send_from! Can we avoid that somehow?
     i = 0
+    counter = 0
     try:
         while True:
             for d in outputs:
-                if d in flow_freely:
+                if d in flow_freely or counter < 1:
```

Comment:
Are by the way sure about this being right? The ` flow_freely` mailboxes are the one that are not required by any other, if we continue here, does it mean we will not be able to read it elsewhere (I might have to read up a bit on the mailbox system)?

https://github.com/AxFoundation/strax/blob/master/strax/processor.py#L102-L116

Reply:
No this is not the case. Although I just notice I did not include it in the tests as I do not check if data is stored afterwards. I will add it. 
In my first attempt I added it to the flow_freely parameters but this does not work due to your mentioned dependencies. However, if you just skip it during the first fetching you prevent the mailboxes from locking each other. I would lie if I say I understand it by 100% but this is what I learned when writing down the "locking-scheme" for a working and non-working case. In addition I processed two background runs and it worked like a charm. All data has been created and is loadable.

Reply:
okay, I must say I don't fully follow this either. If we are are not 100% sure we understand, please don't hold back on adding comments like `preventing is_first from locking to prevent dependencies deadlocks` and adding a few debug statements of which `d` is actually not locked because of `is_first`